lib/system/database/definitions/mysql.rb

```diff
@@ -493,9 +493,7 @@
 
   trigger 'onboardings' do
     <<~SQL
-      IF NEW.account_id <> master_id THEN
-        SET NEW.tenant_id = NEW.account_id;
-      END IF;
+      SET NEW.tenant_id = (SELECT tenant_id FROM accounts WHERE id = NEW.account_id AND tenant_id <> master_id);
```

Comment:
So, all accounts' `tenant_id`s are expected to be set correctly, right?

Reply:
if `tenant_id` is not correct for an account, it would be broken for most everything else :) they are basically the source of truth.